tests/MongoDB.Driver.Core.Tests/Core/Connections/BinaryConnection_CommandEventTests.cs

```diff
@@ -947,12 +994,29 @@ private void ReceiveMessages(params ReplyMessage<BsonDocument>[] messages)
             }
         }
 
-        private static IEnumerable<object[]> GetRedactedCommands()
+        private static BsonDocument ModifyMessageToTriggerConditionToRedact(string commandName, BsonDocument command)
         {
-            var commands = (IEnumerable<string>)typeof(CommandEventHelper)
-                .GetField("__securitySensitiveCommands", BindingFlags.Static | BindingFlags.NonPublic)
-                .GetValue(null);
-            return commands.Select(c => new object[] { c });
+            switch (commandName)
+            {
+                case "isMaster":
+                    command.Add("speculativeAuthenticate", new BsonDocument("db", "authSource"));
+                    break;
+            }
+
+            return command;
         }
     }
-}
\ No newline at end of file
+
+    internal static class CommandEventHelperReflector
+    {
+        public static IReadOnlyDictionary<string, Func<BsonDocument, bool>> __conditionallySecuritySensitiveCommands =>
+            (IReadOnlyDictionary<string, Func<BsonDocument, bool>>)Reflector.GetStaticFieldValue(
+            typeof(CommandEventHelper),
+            "__conditionallySecuritySensitiveCommands");
+
+        public static IReadOnlyCollection<string> __securitySensitiveCommands =>
+            (IReadOnlyCollection<string>)Reflector.GetStaticFieldValue(
+            typeof(CommandEventHelper),
+           "__securitySensitiveCommands");
+    }
```

Comment:
Instead of using reflection to get these private fields (which I'm suggesting you remove) instead reflect on the ShouldRedactMessage:

    internal static class CommandEventHelperReflector
    {
        public static bool ShouldRedactMessage(string commandName, BsonDocument command)
        {
            return (bool)Reflector.InvokeStatic(typeof(CommandEventHelper), nameof(ShouldRedactMessage), commandName, command);
        }
    }

Reply:
Added a new `InvokeStatic` that takes 2 args. Done